### PR TITLE
Initialize Option Editor and create selecterOptionEditor

### PR DESCRIPTION
**Pull request by linweigao**

### CODE REVIEW

File: src/components/FieldOptionEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import * as data from '../data';
+
+interface IProps {
+    field: data.IField;
+    registry: data.FieldRegistry;
+    onChange: (field: data.IField) => void;
```
Comment (asinner) [inline src/components/FieldOptionEditor.tsx:7]: I think we should expose the complete form definition when a field changes so the consumer doesn't need to manually update the fields array.

**Review by asinner — APPROVED** [review]
Approved, but see comment for onChange() method of field option editor for my suggestion

File: src/components/FieldOptionEditor.tsx (hunk 1 above)
Comment (linweigao) [reply src/components/FieldOptionEditor.tsx:7]: Discussed this offline. We may need to think this a bit more.

The questions is whether the FieldOptionEditor should manage the state of all fields.

Comment (linweigao) [reply src/components/FieldOptionEditor.tsx:7]: Let's put it as an open question.

**Pull request merged**